.github/workflows/wasmcloud.yml

```diff
@@ -81,6 +81,40 @@ jobs:
         name: wasmcloud-universal-darwin
         path: wasmcloud-universal-darwin
 
+  oci:
+    runs-on: ubuntu-22.04
+    permissions:
+      packages: write
+    needs: build-bin
+    steps:
+    - uses: actions/checkout@v3
+    - uses: ./.github/actions/install-nix
+      with: 
+        cachixAuthToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
+    - uses: docker/login-action@v2
+      if: ${{ github.event_name != 'pull_request' }}
```

Comment:
I tried searching for the docs. Is this the right page? https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows

If so it looks like there are a lot of event names we don't want to match on. Do we want to run for any other reason than merging to main, i.e. `${{ github.ref == 'refs/heads/main' }}` ?

Reply:
This job can only be triggered by push, PR or manual dispatch. Pushes and dispatches should trigger an upload.

Why would we want to make it harder to test this?

Reply:
What we've done here before is ensure that the job is something that triggers on tag push or manual dispatch, to make an allow list of events that should trigger the action. Based on Connor's documentation link, it does look like there's plenty of other events, how is it guaranteed that other events aren't triggering this?

I think that pushing an artifact with the commit sha on each merge to main / tag push makes sense, I just want to understand that there are not other events that can trigger this (e.g. https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release)

Reply:
The workflow is only triggered by these three events https://github.com/wasmCloud/wasmCloud/pull/598/files#diff-b62bad75f377e25a10e7a10690fb12edea66f96e7df69d1de7873a0324f52b66R4-R11, there is not way any other event (like `release`) could appear here